transaction-status/src/lib.rs

```diff
@@ -367,22 +404,31 @@ impl UiTransactionStatusMeta {
             fee: meta.fee,
             pre_balances: meta.pre_balances,
             post_balances: meta.post_balances,
-            inner_instructions: meta.inner_instructions.map(|ixs| {
-                ixs.into_iter()
-                    .map(|ix| UiInnerInstructions::parse(ix, &account_keys))
-                    .collect()
-            }),
-            log_messages: meta.log_messages,
+            inner_instructions: meta
+                .inner_instructions
+                .map(|ixs| {
+                    ixs.into_iter()
+                        .map(|ix| UiInnerInstructions::parse(ix, &account_keys))
+                        .collect()
+                })
+                .into(),
+            log_messages: meta.log_messages.into(),
             pre_token_balances: meta
                 .pre_token_balances
-                .map(|balance| balance.into_iter().map(Into::into).collect()),
+                .map(|balance| balance.into_iter().map(Into::into).collect())
+                .into(),
             post_token_balances: meta
                 .post_token_balances
-                .map(|balance| balance.into_iter().map(Into::into).collect()),
-            rewards: if show_rewards { meta.rewards } else { None },
-            loaded_addresses: None,
-            return_data: meta.return_data.map(|return_data| return_data.into()),
-            compute_units_consumed: meta.compute_units_consumed,
+                .map(|balance| balance.into_iter().map(Into::into).collect())
+                .into(),
+            rewards: OptionSerializer::Some(
+                if show_rewards { meta.rewards } else { None }.unwrap_or_default(),
+            ),
```

Comment:
The behavior is different here still because rewards used to serialize to null but now serialize to default.